platform/default/src/mbgl/storage/offline_database.cpp

```diff
@@ -1122,16 +1121,25 @@ T OfflineDatabase::getPragma(const char* sql) {
 // delete an arbitrary number of old cache entries. The free pages approach saves
 // us from calling VACUUM or keeping a running total, which can be costly.
 bool OfflineDatabase::evict(uint64_t neededFreeSize) {
+    bool evicted = true;
     uint64_t pageSize = getPragma<int64_t>("PRAGMA page_size");
     uint64_t pageCount = getPragma<int64_t>("PRAGMA page_count");
 
-    auto usedSize = [&] {
+    auto currentDatabaseSize = [&] {
         return pageSize * (pageCount - getPragma<int64_t>("PRAGMA freelist_count"));
     };
 
-    // The addition of pageSize is a fudge factor to account for non `data` column
-    // size, and because pages can get fragmented on the database.
-    while (usedSize() + neededFreeSize + pageSize > maximumAmbientCacheSize) {
+    // Current DB size.
+    const uint64_t initialDatabaseSize = currentDatabaseSize();
+
+    // Current ambient cache size.
+    uint64_t ambientCacheSize = initAmbientCacheSize() == nullptr ? *currentAmbientCacheSize
```

Comment:
s/ambientCacheSize/currentAmbientCacheSize ? than the comment above can be omitted